source/server-selection/server-selection.rst

```diff
@@ -849,7 +849,10 @@ For multi-threaded clients, the server selection algorithm is as follows:
 
 5. If there are any suitable servers, filter them according to `Filtering
    suitable servers based on the latency window`_ and continue to the next step;
-   otherwise, goto Step #9.
+   otherwise, log a `"Waiting for suitable server to become available" message`_
+   if one has not already been logged for this operation OR if the topology
```

Comment:
I was thinking it's the same definition of "changed" that causes us to emit a SDAM monitoring event saying that the topology changed, which I believe happens whenever a server is added, removed, or the new description is not equal to the old one per the equality definition in [SDAM](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#server-description-equality), which omits RTT and some other fields. 
if we want to keep this behavior then I will make that more explicit.

regarding how beneficial this is, @patrickfreed and I discussed this some and thought that it could be useful to be able to identify cases where an operation continued to fail to select a server as the topology changed. I was thinking it would be nice in that it means we automatically get a view into SDAM state changes that are happening during the time period we are waiting for a suitable server. 

on the flipside, if there are a lot of operations going at once, we might end up with a lot of redundant logs since each operation would be re-logging on every topology change. it might be better to just tell the user to turn on SDAM logging instead, so there is a single log message for each topology change.

Reply:
Can you paste the logs from your test somewhere so we can take a look?

Reply:
> I think I would prefer to remove the re-logging aspect here in favor of having users use server selection + SDAM logging in tandem to get the desired effect.

SGTM

Reply:
see [here](https://gist.github.com/kmahar/f3cfdbeef09cf78c6e4e18564511a0e5) for some sample messages running with min severity level info during a failover. I had to re-generate the messages because I had restarted my computer and so lost my terminal history. this time I got slightly more interesting results: you can see in the diff for M1 vs M2 that in M2 the third server has changed its view of who the primary is and also gains an election ID. 
(note: at first glance it looks like the tag sets change, but this is just because we store them in a hashmap and so the ordering for printing out the values is not deterministic.) 

nonetheless, those changes would also be captured in a "topology description changed" log message.

Reply:
I didn't include any debug level messages there since it's quite noisy due to the way my program continually loops (it's hard to time starting the program with the failover starting), but if you want to see any of those let me know

Reply:
> I think I would prefer to remove the re-logging aspect here in favor of having users use server selection + SDAM logging in tandem to get the desired effect.

The examples you linked only contain server selection log messages.  Can you show what they look like in tandem?

Reply:
talked about on slack with Jeff, for posterity: Rust doesn't have SDAM logging yet so that's why those aren't included 
I went ahead and updated the spec so that the "waiting" messages only get logged once and updated tests accordingly